Zend/zend_language_parser.y

```diff
@@ -1228,7 +1228,16 @@ expr:
 			{ $$ = zend_ast_create(ZEND_AST_ASSIGN, $1, $3); }
 	|	variable '=' ampersand variable
 			{ $$ = zend_ast_create(ZEND_AST_ASSIGN_REF, $1, $4); }
-	|	T_CLONE expr { $$ = zend_ast_create(ZEND_AST_CLONE, $2); }
+	|	T_CLONE '(' T_ELLIPSIS ')' {
```

Comment:
So AFAIU, clone is supposed to act like  a normal function. However, named parameters don't work: `clone(object: new stdclass);` is a syntax error.

Reply:
Passing extra arguments (or no arguments) is also not yet supported syntax-wise.

Supporting the entire syntax required non-trivial parser changes, due to the ambiguity of `clone($foo)`. I have a preliminary patch for that, but will need to carefully test it to ensure it behaves as expected in all cases. I plan to follow up with them once I tested them.

Note that the `\clone()` syntax with the explicit backslash already supports all possible variants.

Reply:
> And the goal is for that patch to get into 8.5 ?

Absolutely. The preliminary patch is already part of #18747 (with will add the clone-with support), but I spun out this PR with the “safe” changes to simplify review. The parser changes might or might not become a separate PR.